File: src/types/note.ts
export interface Note {
  id: string;
  workspaceId: string | null;
  folderId: string | null;
  title: string;
  content: string;
  tags: string[];
  isFolder: boolean;
  sortOrder: number;
  createdAt: string;
  updatedAt: string;
}

export interface CreateNoteInput {
  workspaceId?: string | null;
  folderId?: string | null;
  title: string;
  content?: string;
  tags?: string[];
  isFolder?: boolean;
}

export interface UpdateNoteInput {
  id: string;
  title?: string;
  content?: string;
  tags?: string[];
}

File: src-tauri/src/models/note.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Note {
    pub id: String,
    pub workspace_id: Option<String>,
    pub folder_id: Option<String>,
    pub title: String,
    pub content: String,
    pub tags: Vec<String>,
    pub is_folder: bool,
    pub sort_order: i64,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateNoteInput {
    pub workspace_id: Option<String>,
    pub folder_id: Option<String>,
    pub title: String,
    pub content: Option<String>,
    pub tags: Option<Vec<String>>,
    pub is_folder: Option<bool>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateNoteInput {
    pub id: String,
    pub title: Option<String>,
    pub content: Option<String>,
    pub tags: Option<Vec<String>>,
}
